### DISCUSSION TITLE
HomeKey on Homebridge

### DISCUSSION THREAD

**Discussion by VaBanck** [rednblkx/HomeKey-ESP32#46] [General]
Hello everyone, someone set up HomeKey on Homebridge

**Comment by rednblkx** [comment]
Hi there!

Is that supposed to be a question or an affirmation?

Anyhow, right now afaik there is no homekey integration for homebridge apart from a useless dummy one and making one it’s not currently on my list to do.
However, I do have this idea of a project in the back of my mind to eventually do of making the ESP32 a simple NFC reader and offload all of the HomeKit and Homekey stuff elsewhere, anyone interested react with 👀 to this comment.

**Reply by ihrapsa** [reply]
Unless it will improve latency or make it battery efficient, I feel like that would add unnecessary complexity to the lock 🤔 

If that would facilitate  the use of other NFC chips, I might go for it. However, I would much rather go for a stable standalone and battery powered system that could also function when power is down.
 

Have you considered implementing thread/matter?

**Reply by VaBanck** [reply]
I took an example from here, but I can't turn it on https://github.com/arachnetech/homebridge-mqttthing/wiki/Tested-Configurations#lock-mechanism

**Reply by 011V32** [reply]
So I can't connect my lock directly to the ESP32, which means I need a device in between anyway. (In my case this is HomeAssistant)
So for me it doesn't matter where this implementation is located.

**Reply by rednblkx** [reply]
@ihrapsa the idea is that it would be easier to make it battery-powered so you can place it anywhere you have a network connection since I won’t have to deal with all the HomeKit stuff but that’s another project for another day.
For this project, I still intend to make it battery-powered and also entirely standalone, however, that requires a bit of refactoring as I’ll need to leverage BLE which HomeSpan doesn’t do and adding it would overcomplicate the codebase and increase the size.

Regarding Thread, I do have some thoughts on that: 
 - firstly that would require a specific chip with 802.15.4 like ESP32-H2 which I believe would make this project less accessible
 - there is no documentation on how does that work exactly with HomeKit and if HomeKey is supported
 - there have been many reliability issues with Matter and Thread, no idea how much of that has been solved so far but one thing is certain there is much to improve/add, not to be mistaken, I love what matter and thread are trying to do but there is a long road ahead till it becomes a thing.
 
 I might change my mind in the future or not, depends on how things develop, however, matter is something absolutely possible on the current hardware but again I’ll need to find some documentation how does that integrates with HomeKit exactly.

**Reply by rednblkx** [reply]
@VaBanck could you please walk me through what are you trying to do exactly?

**Reply by VaBanck** [reply]
> @VaBanck could you please walk me through what are you trying to do exactly?

  "accessories": [
        {
            "type": "lockMechanism",
            "name": "Name",
            "url": "mqtt",
            "username": "username",
            "password": "*******",
            "mqttPubOptions": {
                "retain": true
            },
            "logMqtt": true,
            "topics": {
                "getCurrenLockState": "set_state",
                "getOnline": {
                    "topic": "state",
                    "apply": "if(message == '1') message='1'; if(message == '0' || message=='0') message='0'; return message;"
                },
                "setOnLineState": "set_state",
                "getLockCurrentState": {
                    "topic": "state",
                    "apply": "if(message == '1') message='1'; if(message == '0' || message=='0') message='0'; return message;"
                },
                "setLockCurrentState": "set_state",
                "getLockTargetState": {
                    "topic": "state",
                    "apply": "if(message == '1') message='1'; if(message == '0' || message=='0') message='0'; return message;"
                },
                "setLockTargetState": "target_state"
            },
            "onlineValue": "set_state",
            "offlineValue": "set_state",
            "lockValues": [
                "0",
                "1",
                "2",
                "3"

When I turn on HomeKey in HomeKit, I want homebridge to send back the command that the lock is open or closed

**Reply by rednblkx** [reply]
First of all please refrain from posting any private information.
Second, I’m sorry but I don’t really see the purpose of this configuration, you seem to just add another layer of HomeKit.
I want to help but I’ll need a bit more context, what exactly is homebridge doing in this whole equation? I don’t see how it is connected to anything else. Also you missed the “t” in “getCurrentLockState”, that might possibly be part of the issue.

**Reply by 011V32** [reply]
Because you mentioned Thread: [Esp32-C6-Bug](https://www.crowdsupply.com/prokyber-s-r-o/esp32-c6-bug)
